sql/src/main/java/org/opensearch/sql/sql/parser/AstExpressionBuilder.java

```diff
@@ -153,6 +156,33 @@ public UnresolvedExpression visitScalarFunctionCall(ScalarFunctionCallContext ct
     return buildFunction(ctx.scalarFunctionName().getText(), ctx.functionArgs().functionArg());
   }
 
+  @Override
+  public UnresolvedExpression visitNestedFunctionCall(NestedFunctionCallContext ctx) {
+    List<UnresolvedExpression> args = new ArrayList();
+
+    args.add(
+        new QualifiedName(
+            ctx.nestedFunction().nestedField().nestedIdent().stream()
+                .map(RuleContext::getText)
+                .map(StringUtils::unquoteIdentifier)
+                .collect(Collectors.toList())
+        )
+    );
+    if (ctx.nestedFunction().nestedPath() != null) {
+      args.add(
+        new QualifiedName(
+            ctx.nestedFunction().nestedPath().nestedIdent().stream()
+                .map(RuleContext::getText)
+                .map(StringUtils::unquoteIdentifier)
+                .collect(Collectors.toList())
+        )
+      );
+    }
+
+    return new Alias(ctx.nestedFunction().nestedField().getText(),
+        new Function(NESTED.getName().getFunctionName(), args));
```

Comment:
To match the legacy implementation the NESTED function is displayed as the FIELD only in the column. So the function call `nested(message.info, message)` is displayed only as `message.info`. This way we can customize the `Alias` to be only the `field` name rather than whole function.